File: SudokuKata/SudokuKata/Utilities/RandomNumber.cs
using System;

namespace SudokuKata.Utilities
{
    public class RandomNumber : IRandomValueGenerator
    {
        private Random rnd;
        public RandomNumber()
        {
            rnd = new Random();
        }

        public RandomNumber(Int32 seed)
        {
            rnd = new Random(seed);
        }

        public int Next()
        {
            var next = rnd.Next();
            // Console.WriteLine(">>> Next random number: {0}.", next);
            return next;
        }

        public int Next(int limit)
        {
            var next = rnd.Next(limit);
            // Console.WriteLine(">>> Next random number: {0} out of {1}.", next, limit);
            return next;
        }
    }
}

File: SudokuKata/SudokuKataCpp/NotYetPorted/RandomNumber.cpp
#include "RandomNumber.h"

namespace SudokuKata
{

    RandomNumber::RandomNumber()
    {
        rnd = new Random();
    }

    RandomNumber::RandomNumber(int seed)
    {
        rnd = new Random(seed);
    }

    int RandomNumber::Next()
    {
        auto next = rnd->Next();
        // Console.WriteLine(">>> Next random number: {0}.", next);
        return next;
    }

    int RandomNumber::Next(int limit)
    {
        auto next = rnd->Next(limit);
        // Console.WriteLine(">>> Next random number: {0} out of {1}.", next, limit);
        return next;
    }
}
